tests/unit/api/test_metadata.py

```diff
@@ -0,0 +1,189 @@
+            mocked_repository_metadata,
+        )
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.get_task_id", lambda: "123"
+        )
+
+        with open(
+            "tests/data_examples/metadata/update-root-payload.json"
+        ) as f:
+            f_data = f.read()
+
+        payload = json.loads(f_data)
+        response = test_client.post(url, json=payload, headers=token_headers)
+
+        assert response.status_code == status.HTTP_202_ACCEPTED
+        assert response.url == f"{test_client.base_url}{url}"
+        assert response.json() == {
+            "message": "Metadata rotation accepted.",
+            "data": {"task_id": "123"},
+        }
+
+    def test_post_metadata_without_bootstrap(
+        self, test_client, monkeypatch, token_headers
+    ):
+        url = "/api/v1/metadata/"
+
+        mocked_check_metadata = pretend.call_recorder(lambda: False)
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.is_bootstrap_done",
+            mocked_check_metadata,
+        )
+
+        with open(
+            "tests/data_examples/metadata/update-root-payload.json"
+        ) as f:
+            f_data = f.read()
+
+        payload = json.loads(f_data)
+        response = test_client.post(url, json=payload, headers=token_headers)
+
+        assert response.status_code == status.HTTP_200_OK
+        assert response.url == f"{test_client.base_url}{url}"
+        assert response.json() == {
+            "detail": {"error": "Metadata rotation requires bootstrap done."}
+        }
+
+    def test_post_metadata_empty_payload(self, test_client, token_headers):
+        url = "/api/v1/metadata/"
+
+        response = test_client.post(url, json={}, headers=token_headers)
+
+        assert response.status_code == status.HTTP_422_UNPROCESSABLE_ENTITY
+        assert response.url == f"{test_client.base_url}{url}"
+        assert response.json() == {
+            "detail": [
+                {
+                    "loc": ["body", "metadata"],
+                    "msg": "field required",
+                    "type": "value_error.missing",
+                },
+            ]
+        }
+
+    def test_post_metadata_invalid_token(self, test_client, monkeypatch):
+        url = "/api/v1/metadata/"
+
+        token_headers = {"Authorization": "Bearer h4ck3r"}
+
+        mocked_check_metadata = pretend.call_recorder(lambda: True)
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.is_bootstrap_done",
+            mocked_check_metadata,
+        )
+
+        mocked_repository_metadata = pretend.stub(
+            apply_async=pretend.call_recorder(lambda *a, **kw: None)
+        )
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.repository_metadata",
+            mocked_repository_metadata,
+        )
+        monkeypatch.setattr(
+            "repository_service_tuf_api.metadata.get_task_id", lambda: "123"
+        )
```

Comment:
Do we need all of that when the token is invalid?
Wouldn't this fail when arguments are evaluated?

Reply:
Good catch